test/robot/replay/client.go

```diff
@@ -44,17 +45,22 @@ func (c *client) replay(ctx context.Context, t *Task) error {
 	if err := c.manager.Update(ctx, t.Action, job.Running, nil); err != nil {
 		return err
 	}
-	output, err := doReplay(ctx, t.Action, t.Input, c.store, c.tempDir)
-	status := job.Succeeded
-	if err != nil {
-		status = job.Failed
-		log.E(ctx, "Error running replay: %v", err)
-	} else if output.CallError != "" {
-		status = job.Failed
-		log.E(ctx, "Error during replay: %v", output.CallError)
-	}
+	for {
```

Comment:
Bail after N failed attempts, and add a sleep between retries (constant, linear, exponential - don't mind).
Could make a nice `task` package utility function:

```go
// Retry repeatedly calls task until task returns a nil error or the number
// attempts reaches maxAttempts. Retry will sleep for retryDelay between retry
// attempts.
func Retry(ctx context.Context, maxAttempts int, retryDelay time.Duration, task Task) error {
	var err error
	for i := 0; i < maxAttempts; i++ {
		if err = task(ctx); err == nil {
			return nil
		}
		time.Sleep(retryDelay)
	}
	return err
}
```